pkg/kubelet/pod_workers.go

```diff
@@ -99,13 +103,20 @@ func newPodWorkers(runtimeCache kubecontainer.RuntimeCache, syncPodFn syncPodFnT
 		workQueue:                 workQueue,
 		resyncInterval:            resyncInterval,
 		backOffPeriod:             backOffPeriod,
+		podCache:                  podCache,
 	}
 }
 
 func (p *podWorkers) managePodLoop(podUpdates <-chan workUpdate) {
 	var minRuntimeCacheTime time.Time
+
 	for newWork := range podUpdates {
 		err := func() (err error) {
+			if p.podCache != nil && !p.podCache.NewerThan(minRuntimeCacheTime) {
```

Comment:
Nope, now it's here to ensure the workers don't sync prematurely. They'd wait until the cache is up-to-date (newer than the end of last sync).

We could pass this to syncPod here if we want. This PR needs some more work after the rest of the PodStatus PRs are merged.